oci/runtime_v2.go

```diff
@@ -28,7 +57,136 @@ func (r *RuntimeV2) Version() (string, error) {
 }
 
 // CreateContainer creates a container.
-func (r *RuntimeV2) CreateContainer(c *Container, cgroupParent string) (err error) {
+func (r *RuntimeV2) CreateContainer(c *Container, cgroupParent string) error {
+	logrus.Debug("Runtimev2.CreateContainer() start")
+	defer logrus.Debug("RuntimeV2.CreateContainer() end")
+
+	// First thing, we need to start the runtime daemon
+	if err := r.startRuntimeDaemon(c); err != nil {
+		return err
+	}
+
+	// Create IO fifos
+	containerIO, err := cio.NewContainerIO(c.ID(),
+		cio.WithNewFIFOs(fifoGlobalDir, c.terminal, c.stdin))
+	if err != nil {
+		return err
+	}
+
+	f, err := os.OpenFile(c.LogPath(), os.O_WRONLY|os.O_APPEND|os.O_CREATE, 0600)
+	if err != nil {
+		return err
+	}
+
+	containerIO.AddOutput("logfile", f, f)
+	containerIO.Pipe()
+
+	r.ctrs[c.ID()] = containerInfo{
+		cio: containerIO,
+	}
+
+	// We can now create the container, interacting with the server
+	request := &task.CreateTaskRequest{
+		ID:       c.ID(),
+		Bundle:   c.BundlePath(),
+		Stdin:    containerIO.Config().Stdin,
+		Stdout:   containerIO.Config().Stdout,
+		Stderr:   containerIO.Config().Stderr,
+		Terminal: containerIO.Config().Terminal,
+	}
+
+	// Create the container
+	if _, err := r.task.Create(r.ctx, request); err != nil {
+		return errdefs.FromGRPC(err)
+	}
+
+	// Spawn a goroutine waiting for the container to terminate. Once it
+	// happens, the container status is retrieved to be updated.
+	go func() {
```

Comment:
Is it reasonable to call this wait() after the calling of task.Start() successfully? otherwise, If the task.Start() failed, this wait() goroutine will never terminate.

Reply:
Yes I was thinking about this change. Probably makes more sense!

Reply:
done